### ISSUE TITLE
[BUG] Myrmex killing hive mates

### ISSUE THREAD

**Issue by ziaodo** [Rebirth-of-the-Night/Rebirth-Of-The-Night#400]
**Describe the bug** Myrmex Queens and other myrmex will attack other hive members

**To Reproduce**
Steps to reproduce the behavior: Either spawn a queen and wait for queen to start producing eggs (she will attack them when they hatch)

Find a wild hive and wait for the eggs it lays to begin hatching.

**Expected behavior**

Myremex should not be attacking myrmex of the same hive. They are supposed to work together. (They are supposed to attack other hives however.)

This causes player made hives to be useless and lessens the challenge of fighting the generated hives as they kill each other. It also makes befriending them impossible after a time as you must give resin to the workers to befriend the hive, but you can't do that if they have all been killed by the rest of the hive. 

**Screenshots**

While I do not have a screenshot of it, it can be seen in ChosenArchitect's latest video (ep 15). From 8:00 - 8:10 you can see the queen and other attacking each other. 
https://www.youtube.com/watch?v=IksmhULUYPg 
**Info (please complete the following information):**
 - Modpack Version: 2.77.5 (My game, not sure which version is being used in the video)
 - Server or Singleplayer: Singleplayer 
 - Version the world was created on: 2.77.4 (My game, not sure which version is being used in the video)

**Additional context**
 They seem to be able to scaffold like other mobs, this change may have broken their ability to recognize other as a member of their hive.

**Comment by democat3457** [comment]
https://github.com/Alex-the-666/Ice_and_Fire/issues/1975

**Comment by renbou** [comment]
Hey! Checked out Ice And Fire's source, seems like its good on their side - there haven't been any bugs in the current 1.12.2 version related to attacks on the workers from other members of the same hive, the code which checks that seems to be fine. This means that the bug could only be introduced by ROTN changes, specifically epic siege in this case. The bug seems to occur when myrmex soldiers etc are given digging capabilities. Seems like this was introduced in 3de668edc852877449fcc575d5c0c39b65c1318d, then 2.77.5 has been released, and after that the bug was silently fixed in f4b7fe04156a004cb7face5fe695e0cf8859f928 and 9b54b80ab33db710755c7c9023c47a5768809859, maybe some later commits as well.
To recap - it is fixed by removing myrmex creatures from the "digging mobs" list in epic siege config.

**Comment by democat3457** [comment]
@Funwayguy esm bug perhaps?

**Comment by renbou** [comment]
Tested more. Maybe I said too soon that it is fixed specifically by removing the myrmexes from the "digging mobs" list, and it seems like actually testing might be incredibly hard since their ai in this version of Ice and Fire is very clunky. However, changing the Epic Siege config myrmex settings to match the newest one, where all of the myrmex are in the blocked ai change list (AI Blacklist) and are removed from the "digging" and "pillaring" list consistently (or so it seems) makes the myrmex soldiers stop attacking the larvae and workers. By looking thru the config files it doesn't seem like any other mods in ROTN are affecting myrmex behaviour, so I'd still say this is the culprit.

**Comment by KaleThyself** [comment]
Checking the AI for Myrmexes in v.3.1.1a yields the results of no hive mates committing familicide. This issue may be closed.

**Comment by democat3457** [comment]
Thanks for your diagnoses @renbou!
